### ISSUE TITLE
Indexing freezes the whole host server then the process gets killed by system

### ISSUE THREAD

**Issue by nicooo7875** [photoprism/photoprism#497]
Hi, 

**First of all** : you guys are doing a great job on photoprism ! keep it up, it's an awesome project !

**Context** : 
- Raspberry pi 4 w/ 4GB RAM & a 4TB usb hdd
- Photoprism running in docker, alongside other containers (nextcloud, bitwarden, searx, traefik)
- photoprism arm64 image is frehly pulled from yesterday evening
- photos are stored on usb hdd, in a nextcloud directory
- nextcloud is used for mobile upload (i know i know)
- photoprism originals folder is bound to the nextcloud photos directory
- currently ~42500 photos in photoprism library

**The issue** : 
1. i drop some pictures in a new directory in the nextcloud photos directory 
2. once finished i switch to photoprism UI, library, index
3. photoprism finds the new directory, starts indexing
4. after 20/30 seconds, everything is frozen (photoprism, other nextcloud webpages, even ssh terminal sessions to the host server)
5. after some time (could be minutes, could be hours), i can access the server again

i do not get anything obvious when typing `docker container logs photoprism`, but i do see in `dmesg` on the host server that photoprism process was killed because the server ran out of memory.

indexing used to work (i have indexed thousands of pictures before) but i run into this error each time now (trying to index the same directory, that is true).

i would be grateful if you could tell me where to look to try to solve this ! i can provide logs if needed.

thanks.

**Comment by lastzero** [comment]
Thank you! :) Please add swap, so that spikes don't kill the process. You may also reduce the number of workers or disable TensorFlow.

Running photoprism next to all the other services requires more than 4 GB of memory for smooth operation. See system requirements in our docs.

**Comment by lastzero** [comment]
See https://docs.photoprism.org/getting-started/:

> We recommend a modern computer with at least 2 cores and 4 GB of memory for running PhotoPrism. Indexing large photo and video collections significantly benefits from fast, local SSD storage and enough memory for caching.
> 
> If you're running out of memory, you can reduce the number of workers in the configuration. As a measure of last resort, you can additionally disable image classification using TensorFlow.

**Comment by lastzero** [comment]
Is there anything more we can do?

**Comment by nicooo7875** [comment]
Sorry, busy times this weekend, I did not get a chance to test with less
workers yet. I will test with 1 or 2 I think and report back, hopefully
within 72 hours. Thanks.

Le mar. 15 sept. 2020 à 11:17, Michael Mayer <notifications@github.com> a
écrit :

**Comment by nicooo7875** [comment]
alright so i run a quick test with only 2 workers and it indexed everything smoothly, UI stayed responsive, and server stayed reachable. the indexing speed is acceptable for me with 2 workers. by checking with `top` on host raspberry pi i still have around 300MB of free RAM when indexing, so i am going to keep it that way for now and monitor what's going on the next time i index.

and everything was explained in the doc.... sorry for that !! and thanks again.

**Comment by nimjor** [comment]
@lastzero sorry to add this this closed issue, but I seem to be having the same/very similar problem but yet unresolved.

Overview of my environment:
- Dell OptiPlex 7040 Micro, 4-core Intel i5, 16GB RAM <-- Proxmox Virtual Environment host
- Guest VM running Ubuntu 20.04, which hosts a few docker services including Photoprism <-- 2 cores and 10GB allocated
- Nextcloud Linux container running on the Proxmox host
- Raspberry Pi 4 8GB set up as NAS to host Nextcloud files over SMB
- Photoprism originals folder points to the Nextcloud top-level Photos directory within the NAS (mounted to the Ubuntu guest VM via fstab)

I've been experimenting with my setup for a while before actually putting all my mobile photos into Nextcloud and then indexing with Photoprism--until recently I had only done maybe a couple hundred.  Now I've synced all the photos from mine and my wife's iPhones to Nextcloud, so a total of close to 5000--not a crazy amount.  

Unfortunately I keep running into an out of memory error while Photoprism indexes these.  It makes it through maybe a few hundred before choking.  I just bumped up the memory allocation for that VM to 10GB yesterday thinking surely that would be adequate.  I also increased the swapfile size to 8GB.  When Photoprism container is stopped or not indexing, this VM uses about 2.5GB total memory.  When indexing starts, I can watch the memory usage steadily climb until it maxes out, and starts consuming swap, and it eats up all of that too until I have to stop the VM.  I also have **PHOTOPRISM_WORKERS: 1** set in my docker-compose file.

<img width="1028" alt="Screen Shot 2021-07-10 at 5 56 11 PM" src="https://user-images.githubusercontent.com/76883392/125206654-ff637780-e24d-11eb-8d28-47a89a31e582.png">

<img width="774" alt="Screen Shot 2021-07-10 at 5 59 33 PM" src="https://user-images.githubusercontent.com/76883392/125206662-068a8580-e24e-11eb-87fd-1ffe336b5dac.png">

Any suggestions?
